File: csharp/hamming/Hamming.cs
using System;
using System.Linq;

public static class Hamming
{
    public static int Compute(string firstStrand, string secondStrand)
    {
        return firstStrand.Zip(secondStrand, (c1, c2) => c1 != c2 ? 1 : 0).Sum();
    }
}

File: typescript/hamming/hamming.ts
class Hamming {
    compute(firstStrand: string, secondStrand: string): number {
        if (firstStrand.length != secondStrand.length) { throw new Error('DNA strands must be of equal length.')}

        //this 'filter' approach cribbed from Github user @ryantriangles (thank you!)
        return firstStrand.split("").filter((value, index) => secondStrand[index] !== value).length;
    }
}
export default Hamming